src/kv/kv_types.h

```diff
@@ -30,12 +31,15 @@ namespace aft
 
 namespace kv
 {
-  // Version indexes modifications to the local kv store. Negative values
-  // indicate deletion
-  using Version = int64_t;
-  static const Version NoVersion = std::numeric_limits<Version>::min();
+  // Version indexes modifications to the local kv store.
+  using Version = uint64_t;
+  static constexpr Version NoVersion = 0u;
 
-  static bool is_deleted(Version version)
+  // DeletableVersion describes the version of an individual key within each
+  // table, which may be negative to indicate a deletion
+  using DeletableVersion = int64_t;
```

Comment:
Possible overflow issues. Does this mean SeqNo is effectively limited to 2^31 and are there checks to ensure this? E.g. `static_cast<DeletableVersion>(v_)` in `src/kv/untyped_map.h` smells fishy to me.

Reply:
Good spot, I'll amend the wrapping check we already had. In general these casts are probably going to remain looking unsafe/unguarded, but they should be temporary (once #1992 is done, `DeletableVersion` can be removed and we no longer need _any_ signed versions).

Reply:
It might be worth bumping up the importance of #1992 post-1.0 then because a 2^31 transaction limit is clearly going to be a problem sooner rather than later.

Reply:
+1. I had only one half of my brain engaged it seems - it's of course 2^63 :-)

Reply:
I've updated the check in `next_version_internal()` to track this limit correctly - and yes its 2^63 rather than 2^31. We will wrap to 0 and log - this will break any service trying where that actually happens, but retains the semantics we've always had. In a few hundred years we can replace `uint64_t` with a bignum type.